tbtc-ts/src/sweep.ts

```diff
@@ -0,0 +1,107 @@
+// @ts-ignore
+import bcoin from "bcoin"
+// @ts-ignore
+import wif from "wif"
+import { BigNumber } from "ethers"
+import {
+  RawTransaction,
+  UnspentTransactionOutput,
+  Client as BitcoinClient,
+} from "./bitcoin"
+
+/**
+ * Sweeps UTXOs by combining all the provided UTXOs and broadcasting a Bitcoin
+ * P2WPKH sweep transaction.
+ * @dev The caller is responsible for ensuring the provided UTXOs are correctly
+ *      formed, can be spent by the wallet and their combined value is greater
+ *      then the fee.
+ * @param utxos - UTXOs to be combined into one output.
+ * @param fee - the value that should be subtracted from the sum of the UTXOs
+ *              values and used as the transaction fee.
+ * @param walletPrivateKey - Bitcoin private key of the wallet.
+ * @param bitcoinClient - Bitcoin client used to interact with the network.
+ * @returns Empty promise.
+ */
+export async function sweepUtxos(
+  utxos: UnspentTransactionOutput[],
+  fee: BigNumber,
+  walletPrivateKey: string,
+  bitcoinClient: BitcoinClient
+): Promise<void> {
+  const utxosWithRaw: (UnspentTransactionOutput & RawTransaction)[] = []
+  for (const utxo of utxos) {
+    const rawTransaction = await bitcoinClient.getRawTransaction(
+      utxo.transactionHash
+    )
+
+    utxosWithRaw.push({
+      ...utxo,
+      transactionHex: rawTransaction.transactionHex,
+    })
+  }
+
+  const transaction = await createSweepTransaction(
+    utxosWithRaw,
+    fee,
+    walletPrivateKey
+  )
+
+  await bitcoinClient.broadcast(transaction)
+}
+
+/**
+ * Creates a Bitcoin P2WPKH sweep transaction.
+ * @param utxos - UTXOs that should be used as transaction inputs.
+ * @param fee - the value that should be subtracted from the sum of the UTXOs
+ *              values and used as the transaction fee.
+ * @param walletPrivateKey - Bitcoin private key of the wallet.
+ * @returns Bitcoin sweep transaction in raw format.
+ */
+export async function createSweepTransaction(
+  utxos: (UnspentTransactionOutput & RawTransaction)[],
+  fee: BigNumber,
+  walletPrivateKey: string
+): Promise<RawTransaction> {
+  const decodedWalletPrivateKey = wif.decode(walletPrivateKey)
+
+  const walletKeyRing = new bcoin.KeyRing({
+    witness: true,
```

Comment:
Have we confirmed that actually produces a P2WPKH transaction? Shouldn't we set something more in the `transaction.addOutput` or `transaction.fund` lines?